src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -542,74 +530,105 @@ class CVulkanLogicalDevice final : public ILogicalDevice
+                for (uint32_t j = 0u; j < pDescriptorWrites[i].count; ++j)
+                    vk_bufferViews.push_back(dummyBufferView);
+            } break;
+
+            case asset::IDescriptor::EC_ACCELERATION_STRUCTURE:
+            {
+                auto& writeAS = vk_writeDescriptorSetAS[i];
+                writeAS = { VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET_ACCELERATION_STRUCTURE_KHR, nullptr };
+                writeAS.accelerationStructureCount = pDescriptorWrites[i].count;
+
+                vk_writeDescriptorSets[i].pNext = &writeAS;
+
+                writeAS.pAccelerationStructures = reinterpret_cast<VkAccelerationStructureKHR*>(vk_accelerationStructures.size());
                 for (uint32_t j = 0u; j < pDescriptorWrites[i].count; ++j)
+                    vk_accelerationStructures.push_back({});
+            } break;
+
+            default:
+                assert(!"Invalid code path.");
+            }
+        }
+
+        for (uint32_t i = 0u; i < descriptorWriteCount; ++i)
+        {
+            switch (pDescriptorWrites[i].info->desc->getTypeCategory())
+            {
+            case asset::IDescriptor::E_CATEGORY::EC_BUFFER:
+            {
+                vk_writeDescriptorSets[i].pBufferInfo = reinterpret_cast<size_t>(vk_writeDescriptorSets[i].pBufferInfo) + vk_bufferInfos.data();
+
+                const auto* infoSrc = pDescriptorWrites[i].info;
+                auto* infoDst = const_cast<VkDescriptorBufferInfo*>(vk_writeDescriptorSets[i].pBufferInfo);
+                for (uint32_t j = 0; j < pDescriptorWrites[i].count; ++j, ++infoSrc, ++infoDst)
                 {
-                    if (pDescriptorWrites[i].info[j].info.buffer.size)
+                    if (infoSrc->info.buffer.size)
                     {
-                        vk_bufferViews[bufferViewOffset + j] = static_cast<const CVulkanBufferView*>(pDescriptorWrites[i].info[j].desc.get())->getInternalObject();
+                        infoDst->buffer = static_cast<const CVulkanBuffer*>(infoSrc->desc.get())->getInternalObject();
+                        infoDst->offset = infoSrc->info.buffer.offset;
+                        infoDst->range = infoSrc->info.buffer.size;
                     }
-                    else
+                }
+            } break;
+
+            case asset::IDescriptor::E_CATEGORY::EC_IMAGE:
+            {
+                vk_writeDescriptorSets[i].pImageInfo = reinterpret_cast<size_t>(vk_writeDescriptorSets[i].pImageInfo) + vk_imageInfos.data();
+
+                const auto* infoSrc = pDescriptorWrites[i].info;
+                auto* infoDst = const_cast<VkDescriptorImageInfo*>(vk_writeDescriptorSets[i].pImageInfo);
+                for (uint32_t j = 0; j < pDescriptorWrites[i].count; ++j, ++infoSrc, ++infoDst)
+                {
+                    if (infoSrc->info.image.imageLayout != asset::IImage::E_LAYOUT::EL_UNDEFINED)
```

Comment:
why do you check this?

Reply:
Same as https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#discussion_r1124149562

Reply:
ok, then same as https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#discussion_r1127710135

Reply:
So continuing on with https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#discussion_r1131372230
we wouldn't have updated the array elements which were not valid (hence assumed to be not dynamically used) and they would've got the dummy values (the first valid array element).

But all of this can also go now.

Reply:
Resolved.